src/v/cluster/members_manager.cc

```diff
@@ -170,13 +171,36 @@ members_manager::apply_update(model::record_batch b) {
     co_return co_await ss::visit(
       cmd,
       [this](decommission_node_cmd cmd) mutable {
-          return dispatch_updates_to_cores(cmd);
+          auto id = cmd.key;
+          return dispatch_updates_to_cores(cmd).then(
+            [this, id](std::error_code error) {
+                if (!error) {
+                    _allocator.local().decommission_node(id);
+                    _update_monitor.notify(node_update{
+                      .id = id, .type = node_update_type::decommissioned});
+                }
+                return error;
+            });
       },
       [this](recommission_node_cmd cmd) mutable {
-          return dispatch_updates_to_cores(cmd);
+          auto id = cmd.key;
+          return dispatch_updates_to_cores(cmd).then(
+            [this, id](std::error_code error) {
+                if (!error) {
+                    _allocator.local().recommission_node(id);
+                    _update_monitor.notify(node_update{
+                      .id = id, .type = node_update_type::recommissioned});
+                }
+                return error;
+            });
       });
 }
 
+ss::future<members_manager::node_update> members_manager::wait_for_node_updates(
+  std::optional<std::reference_wrapper<ss::abort_source>> as) {
+    return _update_monitor.wait(as);
```

Comment:
the abortable monitor keeps update in fifo order, but allows consumers to scale out. in the context of the controller handling updates it seems like processing requests in order is critical. but that suggests that you can't scale out consumers that are handling the updates (otherwise the processing order is unpredictable). that begs the question: can you use a simpler primitive like a ss::queue?

Reply:
I was considering `ss::queue` or even `ss::pipe`, it isn't really easy to express assumption that there may be only one consumer. But sounds good, I think that most of the seastar interfaces makes similar assumptions.